python/dcr_template.py

```diff
@@ -1411,33 +1561,58 @@ def _kernel_1d(offset, size, n_substep=None, lanczos=None):
         if loc % 1.0 == 0:
             kernel[int(loc)] += 1.0
         else:
-            x = pi*(pix - loc)
-            if lanczos is None:
-                kernel += np.sin(x)/x
+            if debug_sinc:
+                i_low = int(np.floor(loc))
+                i_high = i_low + 1
+                frac_high = loc - i_low
+                frac_low = 1. - frac_high
+                kernel[i_low] += frac_low
+                kernel[i_high] += frac_high
             else:
-                kernel += (np.sin(x)/x)*(np.sin(x/lanczos)/(x/lanczos))
+                x = pi*(pix - loc)
+                if lanczos is None:
+                    kernel += np.sin(x)/x
+                else:
+                    kernel += (np.sin(x)/x)*(np.sin(x/lanczos)/(x/lanczos))
     return kernel/n_substep
 
 
 def parallactic_angle(hour_angle, dec, lat):
     """!Compute the parallactic angle given hour angle, declination, and latitude.
 
-    @param hour_angle  Hour angle of the observation, in radians.
-    @param dec  Declination of the observation, in radians.
-    @param lat  Latitude of the observatory, in radians
-    @return  Returns the parallactic angle of the observation, in radians.
+    Parameters
+    ----------
+    hour_angle : lsst.afw.geom.Angle
+        Hour angle of the observation
+    dec : lsst.afw.geom.Angle
+        Declination of the observation.
+    lat : lsst.afw.geom.Angle
+        Latitude of the observatory.
     """
-    return np.arctan2(np.sin(hour_angle), np.cos(dec)*np.tan(lat) - np.sin(dec)*np.cos(hour_angle))
+    y_term = np.sin(hour_angle.asRadians())
+    x_term = (np.cos(dec.asRadians())*np.tan(lat.asRadians()) -
+              np.sin(dec.asRadians())*np.cos(hour_angle.asRadians()))
+    return np.arctan2(y_term, x_term)
 
 
 def wrap_warpExposure(exposure, wcs, BBox, warpingControl=None):
```

Comment:
This is for simplicity and to save memory. I think it's safe with the existing code, but I can imagine that under some future uses or changes it might lead to bugs. I've added ticket DM-9916 to deal with this properly.